src/console_implementation/main_header.hpp

```diff
@@ -5,16 +5,21 @@
 #include <memory>
 #include <thread>
 #include <chrono>
+#include <string>
+#include <fstream>
+#include <sstream>
 
 //Library definitions
 #include "../business_logic/include/Tplayer.hpp"
 #include "Idrawable.hpp"
 #include "drawable_player.hpp"
+#include "../business_logic/include/Tlevel.hpp"
 
 //Global game loop functions
 
 bool input_processing(const std::shared_ptr<Ientity>& player);
-WINDOW* initial_window_gameplay_output();
+WINDOW* initial_window_gameplay_output(std::shared_ptr<Ilevel> p_current_level, unsigned int height, unsigned int width);
 WINDOW* initial_window_scoreboard_output();
 bool gameloop();
 void update(WINDOW* win, std::shared_ptr<drawable_player> player);
+std::string map_converter(std::string file_name, unsigned int* map_height, unsigned int* map_width);
```

Comment:
Basic data types like _int_, _double_, _string_ should be passed by value. It'll use move semantic.
```suggestion
std::string map_converter(std::string file_name, unsigned int map_height, unsigned int map_width);
```
**However**, if it is used to return many variables from function, better practice is to use tuple - https://en.cppreference.com/w/cpp/utility/tuple
```suggestion
std::tuple<std::string, unsigned int, unsigned int> map_converter(std::string file_name);
```
or create struct containing all required values:
```suggestion
struct map_details{
std::string string_representation;
unsigned int height;
unsigned int width;
}
map_details map_converter(std::string file_name);
```
It better describes the intent of the function.

Ref: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f21-to-return-multiple-out-values-prefer-returning-a-struct-or-tuple